lib/services/credentials/credential.service.js

```diff
@@ -254,6 +247,21 @@ s.setScopesForCredential = function (id, type, scopes) {
     }).then(() => true);
 };
 
+function processCredential (credential, options = { includePassword: false }) {
+  if (credential.scopes && credential.scopes.length > 0) {
+    credential.scopes = JSON.parse(credential.scopes);
+  }
+  const credentialModel = config.models.credentials.properties[credential.type];
+  if (!options.includePassword && credentialModel.properties && credentialModel.properties.passwordKey) {
```

Comment:
IMO better idea is to move these conditions to function

Reply:
Can you elaborate a little bit more here? How would you rephrase it?